platform/core.windows/src/org/netbeans/core/windows/WindowManagerImpl.java

```diff
@@ -48,7 +50,7 @@
  * @author Peter Zavadsky
  */
 @org.openide.util.lookup.ServiceProvider(service=org.openide.windows.WindowManager.class)
-public final class WindowManagerImpl extends WindowManager implements Workspace {
+public final class WindowManagerImpl extends WindowManager implements Workspace, WindowManager.ModeManager {
```

Comment:
Unlike `Mode`, `WindowManager` is abstract class - e.g. the methods could be added to it. They would just need some default implementation.

Reply:
Moving methods back into WindowManager from ModeUtilities.